api/v1beta2/topology.go

```diff
@@ -0,0 +1,65 @@
+package v1beta2
```

Comment:
For v1alpha1 (DNSPolicy and TLSPolicy) and v1beta2 (AuthPolicy and RateLimitPolicy) I think this is fine. For v1alpha2/v1beta3 (and therefore v1 as well), let's make sure these methods are defined along with each API type and not in a separate file as a separate concern, please. Reasoning behind is that the `machinery.Policy` interface is what makes the type a policy, not something we need to do just to comply with the Policy Machinery reqs.

Reply:
I am not sure I follow what you are saying here. I will address the other points and then catch up  with you on this point.